src/identity/IdentityProviderHttpHandler.ts

```diff
@@ -1,67 +1,58 @@
-import { OkResponseDescription } from '../ldp/http/response/OkResponseDescription';
 import { RedirectResponseDescription } from '../ldp/http/response/RedirectResponseDescription';
-import type { ResponseDescription } from '../ldp/http/response/ResponseDescription';
+import { ResponseDescription } from '../ldp/http/response/ResponseDescription';
 import type { ResponseWriter } from '../ldp/http/ResponseWriter';
 import type { Operation } from '../ldp/operations/Operation';
 import { BasicRepresentation } from '../ldp/representation/BasicRepresentation';
-import type { RepresentationPreferences } from '../ldp/representation/RepresentationPreferences';
+import type { Representation } from '../ldp/representation/Representation';
 import { getLoggerFor } from '../logging/LogUtil';
-import type { HttpHandlerInput } from '../server/HttpHandler';
-import { HttpHandler } from '../server/HttpHandler';
+import { BaseHttpHandler } from '../server/BaseHttpHandler';
+import type { BaseHttpHandlerArgs } from '../server/BaseHttpHandler';
 import type { HttpRequest } from '../server/HttpRequest';
 import type { HttpResponse } from '../server/HttpResponse';
 import type { RepresentationConverter } from '../storage/conversion/RepresentationConverter';
 import { APPLICATION_JSON } from '../util/ContentTypes';
 import { BadRequestHttpError } from '../util/errors/BadRequestHttpError';
-import { assertError, createErrorMessage } from '../util/errors/ErrorUtil';
-import { InternalServerError } from '../util/errors/InternalServerError';
 import { joinUrl, trimTrailingSlashes } from '../util/PathUtil';
-import { addTemplateMetadata } from '../util/ResourceUtil';
+import { addTemplateMetadata, cloneRepresentation } from '../util/ResourceUtil';
+import { readableToDict } from '../util/StreamUtil';
 import type { ProviderFactory } from './configuration/ProviderFactory';
-import type {
-  Interaction,
-  InteractionHandler,
-  InteractionHandlerResult,
-  InteractionResponseResult,
-} from './interaction/email-password/handler/InteractionHandler';
-import { IdpInteractionError } from './interaction/util/IdpInteractionError';
+import type { Interaction } from './interaction/email-password/handler/InteractionHandler';
+import type { InteractionRoute, TemplatedInteractionResult } from './interaction/routing/InteractionRoute';
 import type { InteractionCompleter } from './interaction/util/InteractionCompleter';
 
-const API_VERSION = '0.1';
-
-/**
- * All the information that is required to handle a request to a custom IDP path.
- */
-export class InteractionRoute {
-  public readonly route: RegExp;
-  public readonly handler: InteractionHandler;
-  public readonly viewTemplates: Record<string, string>;
-  public readonly prompt?: string;
-  public readonly responseTemplates: Record<string, string>;
+const API_VERSION = '0.2';
 
+export interface IdentityProviderHttpHandlerArgs extends BaseHttpHandlerArgs {
+  // BaseHttpHandlerArgs fields aren't copied by Components.js
```

Comment:
I don't understand the comment

Reply:
Make Components.js issue and add URL?

Reply:
Problem is that Components.js is going to complain that we're assigning values to variables that are not defined for this class if we don't copy them here. I vaguely remember some discussions about this that it's quite complex to follow interface inheritance. Will look for an existing issue or create one.